Kernel/FileSystem/FUSE/Inode.cpp

```diff
@@ -250,29 +250,116 @@ ErrorOr<void> FUSEInode::flush_metadata()
     return {};
 }
 
-ErrorOr<void> FUSEInode::add_child(Inode&, StringView, mode_t)
+ErrorOr<void> FUSEInode::add_child(Inode& inode, StringView name, mode_t mode)
 {
-    return ENOTIMPL;
+    dev_t rdev = encoded_device(inode.metadata().major_device, inode.metadata().minor_device);
+    (void)TRY(create_child(name, mode, rdev, {}, {}));
+    return {};
 }
 
-ErrorOr<NonnullRefPtr<Inode>> FUSEInode::create_child(StringView, mode_t, dev_t, UserID, GroupID)
+ErrorOr<NonnullRefPtr<Inode>> FUSEInode::create_child(StringView name, mode_t mode, dev_t rdev, UserID, GroupID)
 {
-    return ENOTIMPL;
+    u32 id = identifier().index().value();
+
+    size_t name_offset = Kernel::is_directory(mode) ? sizeof(fuse_mkdir_in) : sizeof(fuse_mknod_in);
+    auto payload = TRY(KBuffer::try_create_with_size("FUSE: Create child buffer"sv, name_offset + name.length()));
+    memset(payload->data(), 0, name_offset);
+
+    if (Kernel::is_directory(mode)) {
+        fuse_mkdir_in* request = bit_cast<fuse_mkdir_in*>(payload->data());
+
+        request->mode = mode;
+        request->umask = 022; // TODO: Honor any mount flags that should override this.
+    } else {
+        fuse_mknod_in* request = bit_cast<fuse_mknod_in*>(payload->data());
+
+        request->mode = mode;
+        request->rdev = rdev;
+        request->umask = 022; // TODO: Honor any mount flags that should override this.
+    }
+
+    memcpy(payload->data() + name_offset, name.bytes().data(), name.length());
```

Comment:
not sure if thats nicer, but you could also go through `name.bytes().copy_to(payload->bytes().slice(name_offset))`
or the like
But feel free to ignore

(I though we had an API in the other direction as well, but anyway...)

Reply:
Or even better maybe fill this in when you have the typed view,
as ideally that would have a trailing array member to fill this in